zio-http/src/main/scala/zio/http/netty/server/ServerInboundHandler.scala

```diff
@@ -233,6 +213,118 @@ private[zio] final case class ServerInboundHandler(
 
 object ServerInboundHandler {
 
+  private[zio] final class NettyProxyRequest(ctx: ChannelHandlerContext, nettyReq: HttpRequest) extends Request {
```

Comment:
There are a few issues with this one:

1. First off, it extends `Request`, which is a public interface that a user would and should expect to function like a case class (even if it is not a case class). This means a user has an expectation of using `Request` in pure code. Yet, methods like `updateHeaders` and `updateMethod` mutably and in-place modify the fields of the request. If this actually leads to a speed up, then we should investigate why `updateHeaders` and `updateMethod` are called at all in a 'hello world' app: why not just construct with the final headers and method? Why construct and then update?
2. Second, it is using a combination of `lazy val`, which leads to one additional allocation per `val` (`Function0`), together with `AtomicReference`, which is a heavier-weight structure (that must be additionally allocated), whose claim to fame is concurrency, which we don't really need if you buy (1). So this is a lot more overhead than just using `def` for each field.

Reply:
Hmm, removing `lazy val` caused the performance to drop.  Theories?

Reply:
| Benchmark       | Domain Split | No Lazy Vals | Result |
| :-------------- | :----------- | :----------- | :----- |
| 5 @ 256 Conns   | 486750.66    | 448991.47    | -7.76% |
| 5 @ 1024 Conns  | 484600.55    | 455762.56    | -5.95% |
| 5 @ 4096 Conns  | 480459.06    | 436705.48    | -9.11% |
| 5 @ 16384 Conns | 425504.64    | 389968.69    | -8.35% |

Reply:
I restored the lazy vals and reran the benchmarks (for science) and things fall back within reasonable tolerances.

| Benchmark       | Domain Split | Restored Lazy Vals `7dfdda1c` | Result |
| :-------------- | :----------- | :---------------------------- | :----- |
| 5 @ 256 Conns   | 486750.66    | 498811.98                     | 2.48%  |
| 5 @ 1024 Conns  | 484600.55    | 489694.02                     | 1.05%  |
| 5 @ 4096 Conns  | 480459.06    | 480330.47                     | -0.03% |
| 5 @ 16384 Conns | 425504.64    | 436812.13                     | 2.66%  |